packages/commonwealth/server/util/roles.ts

```diff
@@ -0,0 +1,262 @@
+import { Model, Transaction, Op, FindOptions } from 'sequelize';
+import { DB } from 'server/models';
+import { sequelize } from '../database';
+import { Permission } from '../models/role';
+import { RoleAssignmentAttributes } from '../models/role_assignment';
+
+export class RoleInstanceWithPermission {
+  _roleAssignmentAttributes: RoleAssignmentAttributes;
+  chain_id: string;
+  permission: Permission;
+  allow: bigint;
+  deny: bigint;
+
+  constructor(
+    _roleAssignmentInstance: RoleAssignmentAttributes,
+    chain_id: string,
+    permission: Permission,
+    allow: bigint,
+    deny: bigint
+  ) {
+    this._roleAssignmentAttributes = _roleAssignmentInstance;
+    this.chain_id = chain_id;
+    this.permission = permission;
+    this.allow = allow;
+    this.deny = deny;
+  }
+
+  public toJSON(): RoleAssignmentAttributes & {
+    chain_id: string;
+    permission: Permission;
+    allow: bigint;
+    deny: bigint;
+  } {
+    return {
+      ...this._roleAssignmentAttributes,
+      chain_id: this.chain_id,
+      permission: this.permission,
+      allow: this.allow,
+      deny: this.deny,
+    };
+  }
+}
+
+export async function findAllRoles(
+  models: DB,
+  findOptions: FindOptions,
```

Comment:
We should add a type argument here to FindOptions specifying what kind of searches are allowed.

Reply:
adding type as role assignment attributes